kafka/source/config/300-kafkasource.yaml

```diff
@@ -126,7 +126,30 @@ spec:
             serviceAccountName:
               type: string
             sink:
-              type: object
+              anyOf:
+                - type: object
+                  description: "the destination that should receive events."
+                  properties:
+                    ref:
+                      type: object
+                      description: "a reference to a Kubernetes object from which to retrieve the target URI."
+                      required:
+                        - apiVersion
+                        - kind
+                        - name
+                      properties:
+                        apiVersion:
+                          type: string
+                          minLength: 1
+                        kind:
+                          type: string
+                          minLength: 1
+                        name:
+                          type: string
+                          minLength: 1
+                    uri:
+                      type: string
+                      description: "the target URI. If ref is provided, this must be relative URI reference."
```

Comment:
You probably want https://github.com/knative/eventing/blob/86b99cb7741934bca31e6b862356f62c43e57c1b/config/300-apiserversource.yaml#L89-L104 also

Reply:
ok, @grantr will PR the same for couch as well - after these are in, ok ?

Reply:
:+1:

Reply:
I'm on it

Reply:
@lionelvillard I think we should also change `anyOf` to `oneOf`